### PR TITLE
Make plugin compatible with Guava 31+

### PR DESCRIPTION
**Pull request by abayer**
Update core and BOM dependencies, exclude Guava from a number of other dependencies, remove usage of `Joiner`, `CharStreams`, `ByteStreams`, and `ImmutableMap`.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,15 +78,16 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.222.x</artifactId>
-        <version>21</version>
+        <artifactId>bom-2.249.x</artifactId>
+        <!-- TODO weekly -->
+        <version>966.v3857b7c82032</version>
```
Comment (abayer) [inline pom.xml]: @jglick - I lifted this from some other plugin I saw you update, but I'm not sure if this is still the "right" thing to do for the BOM. Got any pointers?

Comment (jglick) [reply pom.xml]: Use `bom-weekly`.

File: pom.xml [hunk 2]
```diff
@@ -165,23 +177,21 @@
...
-      <groupId>com.google.guava</groupId>
-      <artifactId>guava</artifactId>
-      <version>${google.guava.version}</version>
-    </dependency>
     <!-- org.joda.time -->
     <dependency>
       <groupId>joda-time</groupId>
       <artifactId>joda-time</artifactId>
-      <version>2.10.5</version>
+      <version>2.10.10</version>
```
Comment (jglick) [inline pom.xml]: bonus points for rewriting to `java.time`

**Review by alecharp — APPROVED** [review]

**Comment by alecharp** [comment]
@donmccasland could you review this and merge if this is acceptable to you? Thanks.

**Review by raul-arabaolaza — APPROVED** [review]

**Review by raul-arabaolaza — APPROVED** [review]

**Comment by car-roll** [comment]
gentle ping for @donmccasland

**Pull request merged**